src/server/models/answer.js

```diff
@@ -0,0 +1,72 @@
+import { DataTypes } from 'sequelize';
+import SequelizeSlugify from 'sequelize-slugify';
+
+import db from '~/server/database';
+import {
+  generateRandomString,
+  generateHashSecret,
+} from '~/server/services/crypto';
+
+export const types = ['property', 'artwork'];
+
+const Answer = db.define('answer', {
+  id: {
+    type: DataTypes.INTEGER,
+    allowNull: false,
+    primaryKey: true,
+    autoIncrement: true,
+  },
+  slug: {
+    type: DataTypes.STRING,
+  },
+  clientId: {
+    type: DataTypes.STRING,
+    allowNull: false,
+    unique: true,
+    validate: {
+      isAlphanumeric: true,
+    },
+  },
+  chainId: {
+    type: DataTypes.STRING,
+    allowNull: false,
+    unique: true,
+    validate: {
+      isAlphanumeric: true,
+    },
+  },
+  secret: {
+    type: DataTypes.STRING,
+    allowNull: false,
+    unique: true,
+    validate: {
+      isAlphanumeric: true,
+    },
+  },
+  type: {
+    type: DataTypes.ENUM,
+    values: types,
+    allowNull: false,
+  },
+});
+
+Answer.hasOne(require('~/server/model/artwork'), {
+  foreignKey: 'slug',
+  allowNull: true,
+});
+Answer.hasOne(require('~/server/model/property'), {
+  foreignKey: 'slug',
+  allowNull: true,
+});
```

Comment:
`slug` does not have to but *can* technically change over time, which might be a problem as the association is based on a possibly changing foreign key? I see that `clientId` and therefore `slug` is rather constant in this case, but intuitively we could just use `id` here (which is a real constant) and be on the safe side for the future however as implementations can change in the future? Alternatively I would also be fine with `clientId` in case you need the linking to the random string for some reason.

Reply:
yeah, I don't think I've been understanding how we're using slugs - they seem to be being used like ids in the baseController?  And I know postgres ids can also change if we restore the database ... anyway, happy to switch to ids here if you think it's better to just use the `id` here

Reply:
Slugs are manily used for human-friendly URLs (`/api/festivals/123` vs. `/api/festivals/leeds-2019`) so they should mainly be a concert for route matching.

Ids are constistent within a database and after migration one would need to make sure they are kept, which is common to do :+1: